sevntu-checks/src/main/java/com/github/sevntu/checkstyle/SevntuUtil.java

```diff
@@ -76,4 +79,40 @@ public static DetailAST getNextSubTreeNode(DetailAST node, DetailAST subTreeRoot
         return toVisitAst;
     }
 
+    /**
+     * Obtains full type name of first token in node.
+     * @param ast an AST node
+     * @return fully qualified name (FQN) of type, or null if none was found
+     */
+    public static String getTypeNameOfFirstToken(DetailAST ast) {
+        final DetailAST findFirstToken = ast.findFirstToken(TokenTypes.TYPE);
+        final FullIdent ident = CheckUtil.createFullType(findFirstToken);
```

Comment:
Not sure if we should test this variable against NPE, a variable definition should always have a TYPE and IDENT right ?

Reply:
Regression would should if an exception is possible.